src/main/java/com/lothrazar/cyclicmagic/util/UtilPlaceBlocks.java

```diff
@@ -50,54 +50,50 @@ public static boolean placeStateSafe(World world, @Nullable EntityPlayer player,
       return false;
     }
     IBlockState stateHere = null;
-    if (player != null && PermissionRegistry.hasPermissionHere(player, placePos) == false) {
+    if (player != null && !PermissionRegistry.hasPermissionHere(player, placePos)) {
       return false;
     }
-    if (world.isAirBlock(placePos) == false) {
+    if (!world.isAirBlock(placePos)) {
       // if there is a block here, we might have to stop
       stateHere = world.getBlockState(placePos);
       if (stateHere != null) {
         Block blockHere = stateHere.getBlock();
-        if (blockHere.isReplaceable(world, placePos) == false) {
+        if (!blockHere.isReplaceable(world, placePos)) {
           // for example, torches, and the top half of a slab if you click
           // in the empty space
           return false;
         }
         // ok its a soft block so try to break it first try to destroy it
         // unless it is liquid, don't try to destroy liquid
-        //blockHere.getMaterial(stateHere)
+        // blockHere.getMaterial(stateHere)
         if (stateHere.getMaterial() != Material.WATER && stateHere.getMaterial() != Material.LAVA) {
-          boolean dropBlock = true;
           if (!world.isRemote) {
-            world.destroyBlock(placePos, dropBlock);
+            world.destroyBlock(placePos, true);
           }
         }
       }
     }
+    if (placeState.getBlock() instanceof BlockLeaves) { //dont let them decay
+      placeState = placeState.withProperty(BlockLeaves.DECAYABLE, false);
+    }
     boolean success = false;
     try {
       // as soon as i added the try catch, it started never (rarely) happening
-      // we used to pass a flag as third argument, such as '2'
-      // default is '3'
-      // if (!world.isRemote) {
-      if (placeState.getBlock() instanceof BlockLeaves) {//dont let them decay
-        placeState = placeState.withProperty(BlockLeaves.DECAYABLE, false);
+      // flags specifies what to update, '3' means notify client & neighbors
+      // isRemote to make sure we are in a server thread
+      if (!world.isRemote) {
+        success = world.setBlockState(placePos, placeState, 3); // returns false when placement failed
       }
-      success = world.setBlockState(placePos, placeState, 3);
       if (success) {
         UtilSound.playSoundPlaceBlock(world, placePos, placeState.getBlock());
         world.markBlockRangeForRenderUpdate(placePos, placePos.up());
-        world.markChunkDirty(placePos, null);
+        world.markChunkDirty(placePos, null); // passing null here may cause an exception with sponge
```

Comment:
Yeah, null is the tile entity and we don't have one.  If they re crashing on that they need to fix it

Vanilla code doesnt use this and allows null since it is literally
```

 public void markChunkDirty(BlockPos pos, TileEntity unusedTileEntity)
    {
        if (this.isBlockLoaded(pos))
        {
            this.getChunkFromBlockCoords(pos).markDirty();
        }
    }
```

Reply:
yes, and they are as mentioned in my original post about this whole issue, way before the PR by curscascis  https://github.com/PrinceOfAmber/Cyclic/issues/549#issuecomment-343748800
https://github.com/SpongePowered/SpongeCommon/pull/1596